### PR TITLE
feat: add an SSH task

### PR DESCRIPTION
**Pull request by loicmathieu**

### CODE REVIEW

File: src/main/java/io/kestra/plugin/fs/ssh/Command.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+                "port: 22",
+                "username: foo",
+                "password: pass",
+                "command: ls",
+            }
+        )
+    }
+)
+public class Command extends Task implements AbstractVfsInterface, RunnableTask<VoidOutput> {
+    private static final long SLEEP_DELAY = 25L;
```
Comment (tchiotludo) [inline src/main/java/io/kestra/plugin/fs/ssh/Command.java:46]:
```suggestion
    private static final long SLEEP_DELAY = 1L;
```

File: src/main/java/io/kestra/plugin/fs/ssh/Command.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+            var renderedHost = runContext.render(host);
+            var renderedPort = runContext.render(port);
+
+            session = new JSch().getSession(runContext.render(username), renderedHost, Integer.parseInt(renderedPort));
+            session.setPassword(runContext.render(password));
+            session.setConfig("StrictHostKeyChecking", strictHostKeyChecking);
+            session.connect();
+
+            channel = (ChannelExec) session.openChannel("exec");
+            channel.setCommand(runContext.render(command));
```
Comment (tchiotludo) [inline src/main/java/io/kestra/plugin/fs/ssh/Command.java:78]: Can you use multiline commands like on bash?

File: src/main/java/io/kestra/plugin/fs/ssh/Command.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+            channel.setOutputStream(outStream);
+            channel.setErrStream(errStream);
+            channel.connect();
+
+            while (channel.isConnected()) {
+                Thread.sleep(SLEEP_DELAY);
+            }
+
+            outStream.toString().lines().forEach(line -> runContext.logger().info(line));
```
Comment (tchiotludo) [inline src/main/java/io/kestra/plugin/fs/ssh/Command.java:87]: The error & out log will be only send at end and out of order, possible to send it in realtime?
Maybe use this [service ](https://github.com/kestra-io/kestra/blob/9c1c012215445d64d0c9f6d15ac7073836d80731/core/src/main/java/io/kestra/core/tasks/scripts/AbstractBash.java#L318) to have also parsing on the output

File: src/main/java/io/kestra/plugin/fs/ssh/Command.java [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+            if(channel.getExitStatus() != 0) {
+                errStream.toString().lines().forEach(line -> runContext.logger().error(line));
+                throw new RuntimeException("SSH command fails with exit status " + channel.getExitStatus());
```
Comment (tchiotludo) [inline src/main/java/io/kestra/plugin/fs/ssh/Command.java:90]:
```suggestion
                throw new Exception("SSH command fails with exit status " + channel.getExitStatus());
```

**Review by tchiotludo — CHANGES_REQUESTED** [review]

File: src/main/java/io/kestra/plugin/fs/ssh/Command.java (hunk 4 above)
Comment (loicmathieu) [reply src/main/java/io/kestra/plugin/fs/ssh/Command.java:90]: Using `Exception` is  usually seen as a bad smell (well `RuntimeException` also but less important).
We should have a proper PuginRunException or something like this in core to use for such use case.

File: src/main/java/io/kestra/plugin/fs/ssh/Command.java (hunk 2 above)
Comment (loicmathieu) [reply src/main/java/io/kestra/plugin/fs/ssh/Command.java:78]: No, you set a command, you establish a connection, you wait untill the server disconnect.
If you want to execute a script, you had to upload it first (via SFTP for ex), then use the SSH task to execute it.

It's not like a batch, it's a way to execute a command on a remote server.

File: src/main/java/io/kestra/plugin/fs/ssh/Command.java (hunk 1 above)
Comment (loicmathieu) [reply src/main/java/io/kestra/plugin/fs/ssh/Command.java:46]: Well, I can lower it but 1ms seems to low.
An SSH command will  imply a network round trip, a remote session on the server be established, and the command itself. So 10ms or 5ms min seems OK.

File: src/main/java/io/kestra/plugin/fs/ssh/Command.java (hunk 3 above)
Comment (loicmathieu) [reply src/main/java/io/kestra/plugin/fs/ssh/Command.java:87]: Yes, it's possible, I'll use the same TreadLog facility.
To add them to the output, I was hesitant as we now have a way to fetch log so putting all logs on the output seems to be a waste of resources storage for me.

File: src/main/java/io/kestra/plugin/fs/ssh/Command.java (hunk 1 above)
Comment (tchiotludo) [reply src/main/java/io/kestra/plugin/fs/ssh/Command.java:46]: ok I was thinking about second don't know why :face_exhaling:

File: src/main/java/io/kestra/plugin/fs/ssh/Command.java (hunk 2 above)
Comment (tchiotludo) [reply src/main/java/io/kestra/plugin/fs/ssh/Command.java:78]: I want to have a common interface on all bash task, the current one on bash only concat [the command](https://github.com/kestra-io/kestra/blob/0a542a884f1f91ff70cb1d8dd75ba182371f6b55/core/src/main/java/io/kestra/core/tasks/scripts/Bash.java#L131), it's better than a simple string

File: src/main/java/io/kestra/plugin/fs/ssh/Command.java (hunk 1 above)
Comment (loicmathieu) [reply src/main/java/io/kestra/plugin/fs/ssh/Command.java:46]: :)

I'll rename the const to add `MS` at the end ... it's a best practice to always add the unit in the name of such parameter.

File: src/main/java/io/kestra/plugin/fs/ssh/Command.java (hunk 4 above)
Comment (tchiotludo) [reply src/main/java/io/kestra/plugin/fs/ssh/Command.java:90]: `Exception` is better than `RuntimeException` because it's not hidden on the signature

**Pull request merged**